### DISCUSSION TITLE
Error while reading ledger while reading

### DISCUSSION THREAD

**Discussion by wenjun8888** [apache/pulsar#20317] [General]
HEY GUYS:

My pulsar cluster has 3 nodes。
After a disk scale and system restart, i find same data lost.
How the size of storage is 105/102/44 ... I think one of these node got problem
Now all of them can pass bookkeeper shell bookiesanity.

But the logs said:
INFO  org.apache.bookkeeper.client.PendingReadOp - Error: Error while reading ledger while reading L111457 E10933 from bookie: 10.1.69.169:3181
23:19:48.017 [BookKeeperClientWorker-OrderedExecutor-1-0] ERROR org.apache.bookkeeper.proto.PerChannelBookieClient - Read for failed on bookie 10.1.69.169:3181 code EIO

And logs became much bigger...

Is there any way i can rebalance the data or sync data between these three node?
Thank you !

**Comment by hangc0276** [comment]
It looks like some ledger was lost in your BookKeeper cluster.

For new data writing to the BookKeeper cluster in a balanced way, we can set `bookkeeperDiskWeightBasedPlacementEnabled=true`
https://github.com/apache/pulsar/blob/1c813fdaeb9f752b84347b5a62e219dd66e79a26/conf/broker.conf#L1071
